aidb/engine/approx_aggregate_join_engine.py

```diff
@@ -0,0 +1,288 @@
+from typing import Dict, List
```

Comment:
Do we save the outputs of the join conditions in the cache?

Reply:
Currently, I directly use the HttpInferenceService to get outputs within a User Defined Function. I will modify it by switching to the use of BoundInferenceService.